drivers/timer/mchp_xec_rtos_timer.c

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <drivers/system_timer.h>
+#include <sys_clock.h>
+#include <spinlock.h>
+#include <arch/arm/cortex_m/cmsis.h>
+
+#define DEBUG_RTOS_TIMER
+
+#define COUNTER_MAX 0x0fffffffUL
+#define TIMER_STOPPED 0xf0000000UL
+
+#if 0
+#define CYC_PER_TICK (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC	\
+		      / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#else
+
+#define CYC_PER_TICK_M ((CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC * 256)	\
+		      / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+#if (CYC_PER_TICK_M & 0xFF) > 0x7F
+#define CYC_PER_TICK ((CYC_PER_TICK_M >> 8) + 1)
```

Comment:
@scottwcpg This additional definition seems to introduce some deviation. See results from experiment
These are average values measured on logic analyzer when toggling a GPIO

ksleep(ms)	CYC_PER_TICK=33  CYC_PER_TICK=32
200		203.1			196.8
150		152.7			148.1
100		102.5			99.42
50		52.31			50.73
25		27.2			26.39
10		12.14			11.78
5		7.121			6.907
1		3.104			3.02